### PR TITLE
feat:gitlocalize

### LINKED ISSUE #160: missing file: hi-in/uninstall.sh

### PR DESCRIPTION
**Pull request by JarbasAl**
adds support for integrating with Gitlocalize

whenever a PR from gitlocalize bot is merged the sync_translations.py script will run and update the `locale` folder with user submitted strings from GitLocalize platform

also closes #160 (typo)

the script has been run to validate it didnt mess the files, please check the minor formatting changes caused by the script

## Summary by CodeRabbit

- **New Features**
	- Introduced a new workflow for automating translation file synchronization.
	- Added comprehensive localization support for the Open Voice OS installation process in multiple languages, including German, Spanish, French, Hindi, Italian, Dutch, Portuguese, and English.

- **Bug Fixes**
	- Corrected string termination in the tuning script for proper functionality.

- **Documentation**
	- Updated comments in Hindi localization files to English for clarity.

### CODE REVIEW

**Comment by JarbasAl** [comment]
for reference, this was the quick script to parse locale files

```python
import json
import os
from os.path import dirname

FILES = ['misc.sh', 'channels.sh', 'detection.sh', 'features.sh',
         'finish.sh', 'methods.sh', 'profiles.sh', 'satellite.sh',
         'summary.sh', 'telemetry.sh', 'tuning.sh', 'uninstall.sh',
         'welcome.sh']

TRANSLATIONS_FOLDER = f"{dirname(dirname(__file__))}/translations"
LOCALE_FOLDER = f"{dirname(dirname(__file__))}/tui/locales"


def get_key(k, lines, f):
    in_k = False
    val = ""
    for l in lines:
        if l.strip().startswith("#"):
            continue  # comment
        if f == "satellite.sh" and k == "content" and 'content="' in l:
            in_k = True
        elif l.strip() == '"':
            in_k = False
        elif l.startswith("export "):
            in_k = False
        elif l.startswith(f"{k.upper()}="):
            in_k = True
        elif '="' in l:
            in_k = False
        if in_k:
            if f == "satellite.sh" and l.strip() == "$content":
                continue
            l = l.split('="')[-1]
            if l.endswith('"'):
                l = l[:-1]
            val += f"\n{l}"
    return val.strip()


def get_locale(lang):
    DATA = {}
    for f in FILES:
        try:
            with open(f"{LOCALE_FOLDER}/{lang}/{f}", "r") as fi:
                lines = fi.read().split("\n")
        except:
            print(f"missing file: {lang}/{f}")
            continue
        # print(lang, data)
        if f == "satellite.sh":
            DATA[f] = {
                "title_host": get_key("title_host", lines, f),
                "title_port": get_key("title_port", lines, f),
                "title_key": get_key("title_key", lines, f),
                "title_password": get_key("title_password", lines, f),
                "content": get_key("content", lines, f),
                "content_host": get_key("content_host", lines, f),
                "content_port": get_key("content_port", lines, f),
                "content_key": get_key("content_key", lines, f),
                "content_password": get_key("content_password", lines, f),
            }
        elif f == "features.sh":
            DATA[f] = {
                "title": get_key("title", lines, f),
                "content": get_key("content", lines, f),
                "skill_description": get_key("skill_description", lines, f),
                "extra_skill_description": get_key("extra_skill_description", lines, f),
                "gui_description": get_key("gui_description", lines, f)
            }
        elif f == "misc.sh":
            DATA[f] = {
                "ok_button": get_key("ok_button", lines, f),
                "yes_button": get_key("yes_button", lines, f),
                "no_button": get_key("no_button", lines, f),
                "back_button": get_key("back_button", lines, f),
            }
        else:
            DATA[f] = {
                "title": get_key("title", lines, f),
                "content": get_key("content", lines, f),
            }
    return DATA


for lang in os.listdir(LOCALE_FOLDER):
    data = get_locale(lang)
    print(lang, data)
    for f in FILES:
        os.makedirs(f"{TRANSLATIONS_FOLDER}/{lang}", exist_ok=True)
        with open(f"{TRANSLATIONS_FOLDER}/{lang}/strings.json", "w") as f:
            json.dump(data, f, indent=2, ensure_ascii=False)
```

File: translations/fr-fr/strings.json [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  "misc.sh": {
+    "ok_button": "Suivant",
+    "yes_button": "Oui",
+    "no_button": "Non",
+    "back_button": "Retour"
+  },
+  "channels.sh": {
+    "title": "Open Voice OS Installation - Canaux d'installation",
+    "content": "Open Voice OS possède deux canaux d'instalaltion:\n\n  - stable (recommandé mais pas encore disponible)\n  - développment\n\nLa version stable d'Open Voice OS est une version bien testée et fiable, adaptée à un usage quotidien.\n\nLa version de développement d'Open Voice OS est destinée aux développeurs et aux passionnés qui souhaitent expérimenter des fonctionnalités de pointe et contribuer au développement de la plateforme.\n\nVeuillez sélectionner une canal:"
+  },
```
[the remark this answers is not in the document]
Comment (JarbasAl) [reply translations/fr-fr/strings.json]: @goldyfruit please review this commit suggestion

File: translations/fr-fr/strings.json [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    "content_host": "Veuillez saisir l'hôte d'HiveMind Listener (DNS ou adresse IP):",
+    "content_port": "Veuillez entrer le port d'écoute d'HiveMind Listener:",
+    "content_key": "Veuillez saisir la clé d'HiveMind Listener liée au satellite:",
+    "content_password": "Veuillez saisir le mot de passe d'HiveMind Listener lié au satellite:"
+  },
+  "summary.sh": {
+    "title": "Open Voice OS Installation - Résumé",
+    "content": "Vous y êtes presque! Voici un résumé des choix que vous avez effectués pour l'installation d'Open Voice OS:\n\n    - Méthode d'installation:   $METHOD\n    - Canal de déployment:      $CHANNEL\n    - Profil d'installation:    $PROFILE\n    - Interface graphique:      $FEATURE_GUI\n    - Compétences par défaut    $FEATURE_SKILLS\n    - Réglages Raspberry Pi:    $TUNING\n\nLes choix effectués lors du processus d'installation d'Open Voice OS ont été soigneusement étudiés pour adapter notre système à vos besoins et préférences.\n\nEst-ce que tout cela vous semble correct?"
+  },
```
[the remark this answers is not in the document]
Comment (JarbasAl) [reply translations/fr-fr/strings.json]: @goldyfruit please review french suggestions

File: translations/fr-fr/strings.json [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  "tuning.sh": {
+    "title": "Open Voice OS Installation - Réglages pour Raspberry Pi",
+    "content": "Open Voice OS fournit un profil de réglages pour les appareils de type Raspberry Pi afin d'optimiser au mieux le système d'exploitation pour recevoir Open Voice OS.\n\nCe processus de réglage vise à améliorer les performances, à réduire l'utilisation des ressources et à garantir une expérience utilisateur plus fluide.\n\nAppliquer les réglages pour Raspberry Pi (recommandé)?"
+  },
+  "uninstall.sh": {
+    "title": "Open Voice OS Installation - Désinstallation",
+    "content": "Une instance d'Open Voice OS a été détectée.\n\nÉtant donné que Docker et PipeWire peuvent avoir été installés par le système ou manuellement, l'ínstallateur ne supprimera pas les packages suivants:\n\n  - docker-ce\n  - docker-compose-plugin\n  - docker-ce-rootless-extras\n  - docker-buildx-plugin\n  - pipewire\n  - pipewire-alsa\n\nVoulez-vous désinstaller Open Voice OS?"
+  },
```
[the remark this answers is not in the document]
Comment (JarbasAl) [reply translations/fr-fr/strings.json]: @goldyfruit please review french suggestions

File: translations/fr-fr/strings.json [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    "title_host": "Open Voice OS Installation - Satellite 1/4",
+    "title_port": "Open Voice OS Installation - Satellite 2/4",
+    "title_key": "Open Voice OS Installation - Satellite 3/4",
+    "title_password": "Open Voice OS Installation - Satellite 4/4",
+    "content": "En se connectant à HiveMind Listener, les satellites HiveMind accèdent à un réseau de connaissances et de capacités partagées, facilitant ainsi une expérience d'assistance vocale unifiée et efficace.",
+    "content_host": "Veuillez saisir l'hôte d'HiveMind Listener (DNS ou adresse IP):",
+    "content_port": "Veuillez entrer le port d'écoute d'HiveMind Listener:",
+    "content_key": "Veuillez saisir la clé d'HiveMind Listener liée au satellite:",
+    "content_password": "Veuillez saisir le mot de passe d'HiveMind Listener lié au satellite:"
+  },
```
[the remark this answers is not in the document]
Comment (JarbasAl) [reply translations/fr-fr/strings.json]: @goldyfruit please review french suggestions

**Review by goldyfruit — APPROVED** [review]

**Pull request merged**